pkg/sql/schemachanger/scplan/internal/rules/dep_index_and_column.go

```diff
@@ -784,6 +812,29 @@ func init() {
 				}),
 			}
 		})
+
+	// We need to ensure that the temporary index has all the relevant writes
+	// to any columns it contains. We ensure elsewhere that any index which
+	// will later be merged with the temporary index is not backfilled until
+	// that temporary index is receiving writes. This rule ensures that those
+	// write operations contain data for all columns.
+	registerDepRule(
+		"column is WRITE_ONLY before temporary index is WRITE_ONLY",
+		scgraph.Precedence,
+		"column", "index",
+		func(from, to nodeVars) rel.Clauses {
+			return rel.Clauses{
+				from.el.Type((*scpb.Column)(nil)),
+				to.el.Type((*scpb.TemporaryIndex)(nil)),
+				indexContainsColumn(to.el, from.el, "index-column", "table-id", "column-id", "index-id"),
+				targetStatus(from.target, scpb.ToPublic),
+				targetStatus(to.target, scpb.Transient),
+				currentStatus(from.node, scpb.Status_WRITE_ONLY),
+				currentStatus(to.node, scpb.Status_WRITE_ONLY),
+			}
+		},
+	)
+
```

Comment:
This rule seems to be relevant when adding a column (so we insist this newly added column reaches WRITE_ONLY before the temporary reaches WRITE_ONLY). I'm curious how you discover the need to add such a dep rule when working on `DROP COLUMN`

Reply:
You're right that this is unrelated to drop column. I don't remember where this came up, but there was a case I ran into where this mattered. I think it was that I removed the `WRITE_ONLY` state from `"column depends on primary index"` because it broke rollbacks of drop column, at which point in the add column case now it was valid to defer moving the column to `WRITE_ONLY` until later which caused some other problem.